addons/project/models/project_task.py

```diff
@@ -993,6 +993,16 @@ def _ensure_fields_are_accessible(self, fields, operation='read', check_group_us
                     error_message = _('You cannot write on the following fields on tasks: %(field_list)s', field_list=unauthorized_field_list)
                 raise AccessError(error_message)
 
+    def _has_field_access(self, field, operation):
+        # additional checks for portal users
+        if super()._has_field_access(field, operation) or not self.env.user._is_portal():
+            return True
+        if operation == 'read':
+            return field.name in self.SELF_READABLE_FIELDS
+        if operation == 'write':
+            return field.name in self.SELF_WRITABLE_FIELDS
+        return False
```

Comment:
This is not correct.  In this method, the non-portal users have access to all fields.  The former implementation is doing the usual checks for all users, and adding `_ensure_fields_are_accessible` for portal users.
```suggestion
    def _has_field_access(self, field, operation):
        if not super()._has_field_access(field, operation):
            return False
        if not self.env.su and self.env.user._is_portal():
            # additional checks for portal users
            if operation == 'read':
                return field.name in self.SELF_READABLE_FIELDS
            if operation == 'write':
                return field.name in self.SELF_WRITABLE_FIELDS
        return True
```